### PR TITLE
Improve cleanup of the FlutterView's Mojo service providers

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: sky/shell/platform/android/io/flutter/view/FlutterView.java [hunk 1]
```diff
@@ -190,6 +193,12 @@ float getDevicePixelRatio() {
     }
 
     public void destroy() {
+        mPlatformServiceProviderBinding.unbind().close();
+        mPlatformServiceProvider.destroy();
```
Comment (abarth) [inline sky/shell/platform/android/io/flutter/view/FlutterView.java]: Do we need to null-check these?  Is it possible that we're told to destroy before we've created them?

File: sky/shell/platform/android/io/flutter/view/FlutterView.java [hunk 2]
```diff
@@ -365,16 +374,12 @@ private void attach() {
...
-                ServiceProvider.MANAGER.getInterfaceRequest(core);
-        ServiceProvider.MANAGER.bind(impl, serviceProvider.second);
-        return serviceProvider.first;
-    }
-
     public void runFromBundle(String bundlePath, String snapshotPath) {
-        if (mPlatformServiceProvider != null) {
-            mPlatformServiceProvider.close();
+        if (mPlatformServiceProviderBinding != null) {
+            mPlatformServiceProviderBinding.unbind().close();
```
Comment (abarth) [inline sky/shell/platform/android/io/flutter/view/FlutterView.java]: We should probably call destroy on the old ones here too.

**Comment by abarth** [comment]
LGTM

File: sky/shell/platform/android/io/flutter/view/FlutterView.java (hunk 1 above)
Comment (jason-simmons) [inline sky/shell/platform/android/io/flutter/view/FlutterView.java]: done

File: sky/shell/platform/android/io/flutter/view/FlutterView.java (hunk 2 above)
Comment (jason-simmons) [inline sky/shell/platform/android/io/flutter/view/FlutterView.java]: done

**Pull request merged**